src/source_sampling.cpp

```diff
@@ -139,14 +144,18 @@ pyne::SourceParticle pyne::Sampler::particle_birth(std::vector<double> rands) {
   xyz_rands.push_back(rands[3]);
   xyz_rands.push_back(rands[4]);
   moab::CartVect pos = sample_xyz(ve_idx, xyz_rands);
-  // cell_number
   if (ve_type == moab::MBHEX) {
-  cell_id = cell_number[ve_idx*max_num_cells + c_idx];
-  } else {
-     cell_id = -1;
+     if (mesh_mode == SUBVOXEL) {
+         cell_list.emplace_back(cell_number[ve_idx*max_num_cells + c_idx]);
+     }
+     else { // Voxel
+        for (int c=0; c<max_num_cells; c++) {
```

Comment:
For voxel mode it would probably be best to maintain a static cell_list rather than create this vector each time we sample a particle.